main.c

```diff
@@ -1657,6 +1665,53 @@ static struct PyModuleDef gmp_module = {
 };
 
 
+#define max_alloc 8192
+static void* mem_arr[max_alloc];
+size_t idx = 0;
+
+
+static void *
+my_allocate_function(size_t size)
+{
+    if (idx > max_alloc) {
+        goto err;
+    }
+
```

Comment:
Please add this
```
if (size == 0) {
    size = 1;
}
```
`malloc` does not handle size `0` internally, it can lead to uncontrollable memory behavior.

Reply:
I think it's fine.  These functions are internal.